src/include/stir/listmode/CListRecordROOT.h

```diff
@@ -28,134 +28,45 @@
-    void init_from_data(const int &_ring1, const int &_ring2,
-                             const int &crystal1, const int &crystal2);
-
-    inline bool is_prompt() const
-    { return true; }
-
-    bool inline is_swapped() const
-    { return swapped; }
-
-private:
-    //! First ring, in order to detector tangestial index
-    int ring1;
-    //! Second ring, in order to detector tangestial index
-    int ring2;
-    //! First detector, in order to detector tangestial index
-    int det1;
-    //! Second detector, in order to detector tangestial index
-    int det2;
-    //! Indicates if swap segments
-    bool swapped;
-    //! This is the number of detector we have to rotate in order to
-    //! align GATE and STIR.
-    int quarter_of_detectors;
-};
-
-//! A class for storing and using a timing 'event' from a listmode file from the ECAT 8_32bit scanner
-/*! \ingroup listmode
- */
-class CListTimeROOT : public CListTime
-{
-public:
-    void init_from_data(double time1, double time2)
-    {
-        timeA = time1;
-        timeB = time2;
-    }
-
-    //! Returns always true
-    bool is_time() const
-    { return true; }
-    //! Returns the detection time of the first photon
-    //! in milliseconds.
-    inline unsigned long  get_time_in_millisecs() const
-    { return timeA * 1e3; }
-    //! Get the detection time of the first photon
-    //! in milliseconds
-    inline double get_timeA_in_millisecs() const
-    { return timeA * 1e3; }
-    //! Get the detection time of the second photon
-    //! in milliseconds
-    inline double get_timeB_in_millisecs() const
-    { return timeB * 1e3; }
-    //! Get the delta Time between the two events
-    inline double get_delta_time_in_millisecs() const
-    { return (timeB - timeA) * 1e3; }
-    //! Get delta time in picoseconds
-    inline  double get_delta_time_in_picosecs() const
-    { return (timeB - timeA) * 1e12; }
-    inline Succeeded set_time_in_millisecs(const unsigned long time_in_millisecs)
-    {
-        warning("set_time_in_millisecs: Not implemented for ROOT files. Aborting.");
-        return Succeeded::no;
-    }
-
-private:
-
-    //!
-    //! \brief timeA
-    //! \details The detection time of the first of the two photons, in seconds
-    double timeA;
-
-    //!
-    //! \brief timeB
-    //! \details The detection time of the second of the two photons
-    double timeB;
-};
+using namespace std;
 
 //! A class for a general element of a listmode file for a Siemens scanner using the ROOT files
 class CListRecordROOT : public CListRecord // currently no gating yet
 {
 public:
     //! Returns always true
-    bool inline is_time() const;
+    bool is_time() const;
     //! Returns always true
-    bool inline is_event() const;
+    bool is_event() const;
     //! Returns always true
-    bool inline is_full_event() const;
+     inline bool is_full_event() const;
 
-    virtual CListEventROOT&  event()
+    virtual CListEvent&  event()
```

Comment:
better to keep this as it was (and all similar functions below) I believe. This should fall under the "covariant return types" rule for virtual functions a far as I understand. (if it compiled anyway...)

Reply:
reverted. But I have to be honest I cannot understand how this works.